proto/src/lib.rs

```diff
@@ -41,19 +40,22 @@ impl Decoder for LdapCodec {
     fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         // How many bytes to consume?
         let mut parser = Parser::new();
-        let (size, msg) = match *parser.handle(Input::Element(buf)) {
-            ConsumerState::Continue(_) => return Ok(None),
-            ConsumerState::Error(_e) => {
-                return Err(io::Error::new(io::ErrorKind::Other, "lber parser"))
+        let (rem, msg) = match parser.parse(buf) {
+            Ok(r) => r,
+            // Need more data
+            Err(nom::Err::Incomplete(want)) => {
+                trace!(?want, "incomplete");
+                return Ok(None);
+            }
+            Err(e) => {
+                error!(err = ?e, "lber parser");
+                return Err(io::Error::new(io::ErrorKind::Other, "lber parser"));
             }
-            ConsumerState::Done(size, ref msg) => (size, msg),
-        };
-        // Consume that
-        let size = match size {
-            Move::Await(_) => return Ok(None),
-            Move::Seek(_) => return Err(io::Error::new(io::ErrorKind::Other, "lber seek")),
```

Comment:
What is the equivalent of "lber seek" in the new code? Can you still get into that path?

Reply:
No, not any more. Because it's nom there isn't a "seek" backwards, it just progresses linearly. So the only cases are now invalid asn.1, and "need more data" because we have incomplete frames. I have already tested the incomplete handling is correct :)